src/main/java/sonia/scm/mail/spi/DefaultMailService.java

```diff
@@ -390,7 +401,8 @@ public void send() throws MailSendBatchException {
 
     private Email createMail(Recipient recipient) {
       Email email = new Email();
-      email.setFromAddress(envelopeBuilder.fromDisplayName, envelopeBuilder.configuration.getFrom());
+      String fromAddress = !Strings.isNullOrEmpty(envelopeBuilder.fromAddress) ? envelopeBuilder.fromAddress : envelopeBuilder.configuration.getFrom();
```

Comment:
I'd move this logic to `EnvelopeBuilderImpl` like this:

```java
    String effectiveFromAddress() {
      return !Strings.isNullOrEmpty(fromAddress) ? fromAddress : configuration.getFrom();
    }
```

This would prevent "misusage", when the property is used in other places.